pandas/tests/io/json/test_readlines.py

```diff
@@ -387,12 +387,13 @@ def to_json_append_output_different_columns():
     df3 = DataFrame({"col2": ["e", "f"], "col3": ["!", "#"]})
     df4 = DataFrame({"col4": [True, False]})
 
+    # Booleans here converted to integers to agree with data created by "to_json"
```

Comment:
this is because missing values in `'col4'` get filled with `np.nan`, which in `numpy` is of dtype `float`
I don't think we need the comment though, int upcasting to float is fairly common around the pandas codebase

Rather than `int(True)`, how about
```python
    expected = DataFrame(
        {
            "col1": [1, 2, 3, 4, None, None, None, None],
            "col2": ["a", "b", "c", "d", "e", "f", None, None],
            "col3": [None, None, None, None, "!", "#", None, None],
            "col4": [None, None, None, None, None, None, True, False],
        }
    ).astype({"col4": "float"})
```

Reply:
Okay, great, thank you, changing it now!

Reply:
Done, the checks are processing again now.

Reply:
should be good thanks

you can ignore the "Doc Build and Upload" failure, it's unrelated